buildSrc/src/main/kotlin/UnpackAar.kt

```diff
@@ -0,0 +1,89 @@
+package app.cash.paparazzi.gradle
+
+import org.gradle.api.Project
+import org.gradle.api.artifacts.dsl.DependencyHandler
+import org.gradle.api.artifacts.transform.InputArtifact
+import org.gradle.api.artifacts.transform.TransformAction
+import org.gradle.api.artifacts.transform.TransformOutputs
+import org.gradle.api.artifacts.transform.TransformParameters
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.attributes.LibraryElements
+import org.gradle.api.file.FileSystemLocation
+import org.gradle.api.provider.Provider
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
+import java.io.File
+import java.nio.file.Files
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+
+// Attributes used by aar dependencies
+private val artifactType = Attribute.of("artifactType", String::class.java)
+private val unpackedAar = Attribute.of("unpackedAar", Boolean::class.javaObjectType)
+
+fun Project.configureAar() =
+  configurations.configureEach {
+    // Causes issue with spotless
+    if (name.contains("spotless", ignoreCase = true)) return@configureEach
+
+    afterEvaluate {
+      if (isCanBeResolved && !isCanBeConsumed) {
+        attributes.attribute(unpackedAar, true) // request all AARs to be unpacked
+      }
+    }
+  }
+
+fun DependencyHandler.configureAarUnpacking() {
+  attributesSchema {
+    attribute(unpackedAar)
+    attribute(KotlinPlatformType.attribute) {
+      compatibilityRules.add(AndroidJvmAndJvmCompatible::class.java)
+    }
+    attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE) {
+      compatibilityRules.add(AarAndJarCompatible::class.java)
+    }
+  }
+
+  artifactTypes {
+    create("aar") {
+      attributes.attribute(unpackedAar, false)
+    }
+  }
+
+  registerTransform(UnpackAar::class.java) {
+    from
+      .attribute(unpackedAar, false)
+      .attribute(KotlinPlatformType.attribute, KotlinPlatformType.androidJvm)
+      .attribute(artifactType, "aar")
+    to
+      .attribute(unpackedAar, true)
+      .attribute(KotlinPlatformType.attribute, KotlinPlatformType.jvm)
+      .attribute(artifactType, "jar")
+  }
+}
+
+@Suppress("UnstableApiUsage")
+abstract class UnpackAar : TransformAction<TransformParameters.None> {
+
+  @get:InputArtifact
+  abstract val inputArtifact: Provider<FileSystemLocation>
+
+  override fun transform(outputs: TransformOutputs) {
+    ZipFile(inputArtifact.get().asFile).use { zip ->
+      zip.entries().asSequence()
+        .filter { !it.isDirectory }
+        .filter { it.name.endsWith(".jar") }
+        .forEach {
+          val outputCacheDirectory = outputs.file(it.name).toString().split("caches/")[1]
```

Comment:
I'm a bit nervous about this as opposed to a local folder in `layout.buildDirectory`.